### ISSUE TITLE
Building errors on macOS

### ISSUE THREAD

**Issue by JihoYang** [precice/precice#139]
Hi!

I had some small issue when building preCICE on macOS. I managed to get it built with very minor fixes but I'm opening an issue to let you guys know of it. 

```
OS: macOS High Sierra 10.13.6

Package Manager used: Homebrew

preCICE version: v1.2.0

gcc: gcc-8 (Homebrew GCC 8.2.0) 8.2.0

(Relevant) Dependencies: Boost 1.67
```

.bash_profile
```
# GCC
export PATH="/usr/local/Cellar/gcc/8.2.0:$PATH"
alias gcc='gcc-8'
alias g++='g++-8'

# preCICE
export PRECICE_ROOT="/Users/Jiho/github/precice"
export DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$PRECICE_ROOT/build/last/

# BOOST
BOOST_ROOT=/usr/local/Cellar/boost/1.67.0_1
export LIBRARY_PATH=$BOOST_ROOT/lib:$LIBRARY_PATH
export LD_LIBRARY_PATH=$BOOST_ROOT/lib:$LD_LIBRARY_PATH
export CPLUS_INCLUDE_PATH=$BOOST_ROOT/include:$CPLUS_INCLUDE_PATH

# Eigen
export CPLUS_INCLUDE_PATH="/usr/local/Cellar/eigen/3.3.5/include/eigen3/:$CPLUS_INCLUDE_PATH"

# PETSc
export PETSC_DIR=/usr/local/Cellar/petsc/3.9.3/
export PETSC_ARCH=macx
```

I ingored `PETSC_ARCH` since `PETSC_DIR` didn't have any directory inside as such (it just has `lib`, `include`, etc).

When I try to build precice (scons petsc=yes python=yes -j 4) it manages to find all the libraries but fails to find lboost_thread when linking.

```
Jihos-MBP:precice Jiho$ scons python=yes petsc=yes -j 4
scons: Reading SConscript files ...

(default)  builddir   = build      Directory holding build files. ( /path/to/builddir )
(default)  build      = Debug      Build type (release|debug|Release|Debug)
(default)  libprefix  = /usr       Path prefix for libraries ( /path/to/libprefix )
(default)  compiler   = mpicxx     Compiler to use.
(default)  mpi        = True       Enables MPI-based communication and running coupling tests. (yes|no)
(modified) petsc      = True       Enable use of the Petsc linear algebra library. (yes|no)
(modified) python     = True       Used for Python scripted solver actions. (yes|no)
(default)  gprof      = False      Used in detailed performance analysis. (yes|no)
(default)  platform   = none       Special configuration for certain platforms (none|supermuc|hazelhen)

Checking whether the C++ compiler works... (cached) yes
Checking for C++ library pthread... (cached) yes
(modified) PETSC_DIR  = /usr/local/Cellar/petsc/3.9.3/
(default)  PETSC_ARCH =         
Checking for C++ library petsc... (cached) yes
Checking for C++ header file Eigen/Dense... (cached) yes
Checking for C++ library boost_log... (cached) yes
Checking for C++ library boost_log_setup... (cached) yes
Checking for C++ library boost_thread... (cached) yes
Checking for C++ library boost_system... (cached) yes
Checking for C++ library boost_filesystem... (cached) yes
Checking for C++ library boost_program_options... (cached) yes
Checking for C++ library boost_unit_test_framework... (cached) yes
Checking for C++ header file boost/vmd/is_empty.hpp... (cached) yes
Checking for C++ header file boost/geometry.hpp... (cached) yes
Checking for C++ header file boost/signals2.hpp... (cached) yes
Checking for C++ header file mpi.h... (cached) yes
(default)  PRECICE_PYTHON_LIB = python2.7
(default)  PRECICE_PYTHON_LIB_PATH = /usr/local/opt/python@2/Frameworks/Python.framework/Versions/2.7/lib
(default)  PRECICE_PYTHON_INC_PATH = /usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/include/python2.7
(default)  PRECICE_NUMPY_INC_PATH = /usr/local/lib/python2.7/site-packages/numpy/core/include
Checking for C++ library python2.7... (cached) yes
Checking for C++ header file numpy/arrayobject.h... (cached) yes
Checking for C++ library xml2... (cached) yes

Targets:   build/debug/libprecice.dylib, build/debug/testprecice, symlink
Buildpath: build/debug
()
scons: done reading SConscript files.
scons: Building targets ...

...............

ld: warning: directory not found for option '-Lpython2.7'
ld: library not found for -lboost_thread
clang: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [build/debug/libprecice.dylib] Error 1
scons: building terminated because of errors.
```

I checked that I have `libboost_thread-mt.dylib` and `libboost_thread-mt.a` in the right directory ($BOOST_ROOT/lib) but it somehow fails to link (perhaps this isn't the right library?)

As a quick workaround I compiled Boost from the source code (version 1.66) as described in the [wiki](https://github.com/precice/precice/wiki/Dependencies#boost) and it successfully found all the Boost libraries but gave the following error:

```
src/com/CommunicationFactory.hpp:18:23: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
  virtual std::string addressDirectory()
                      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iosfwd:193:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
```

I thought this could be a compiler problem so I also tried it with Xcode distributed gcc compiler and I got the same errors. This could be solved simply by adding `#include<string>` in `src/com/CommunicationFactory.hpp`

After this point everything went through smoothly.

Thanks for all the great work!

Jiho

**Comment by jy1737** [comment]
Another thing, 

although preCICE gets built successfully it throws out the following two warnings:

ld: warning: directory not found for option '-Lpython2.7'
ld: warning: directory not found for option '-L/usr/local/lib//lib'

**Comment by fsimonis** [comment]
Thank you for opening the issue!
I fixed the include in aa01fda9ef6632ecedebadd33b9bdb9117d208b3.
